clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -1222,6 +1222,16 @@ static IsTupleLike isTupleLike(Sema &S, SourceLocation Loc, QualType T,
   if (E.isInvalid())
     return IsTupleLike::Error;
 
+  if (Size < 0 || Size >= UINT_MAX) {
+    llvm::SmallVector<char, 16> Str;
+    Size.toString(Str);
```

Comment:
If we don't format the value and it's lower than `UINT_MAX` anyway, just calling `getExtValue()` should work?